Inc/info.h

```diff
@@ -0,0 +1,103 @@
+/**
+@file vport.h
+@brief Generates the device info blob and writes it over serial.
+*/
+
+#pragma once
+#include "TASRun.h"
+#include <string.h>
+#include <stdlib.h>
+
+/*****Preprocessor shenanigans*****/
+#define STRINGIFY2( x) #x
+#define STRINGIFY(x) STRINGIFY2(x)
+#define ARRLEN(x) (sizeof(x) / sizeof(x[0]))
+
+/*****This should be defined by the makefile at compile-time.*****/
+#ifndef GIT_HASH
+#define GIT_HASH No Git hash defined at compile time. Are teehats to blame?
```

Comment:
...teahats?

Reply:
Heh, well spotted - this is a tiny easter egg that spawned from a silly Radio TASBot ad we recorded for dwangoAC's stream. Incidentally, it is also the failsafe version that is reported if no Git hash is defined. This will only show up in a binary if the build system fails to define the macro `GIT_HASH`.